code/datums/elements/can_shatter.dm

```diff
@@ -53,9 +53,11 @@
 
 /// Handles the actual shattering part, throwing shards of whatever is defined on the component everywhere
 /datum/element/can_shatter/proc/shatter(atom/movable/source, atom/hit_atom)
-	var/generator/scatter_gen = generator(GEN_CIRCLE, 0, 48, NORMAL_RAND)
-	var/scatter_turf = get_turf(hit_atom)
+	var/turf/scatter_turf = get_turf(hit_atom)
+	if(!hit_atom.CanPass(source, get_dir(source, hit_atom))) //Object is too dense to fall apart on
+		scatter_turf = get_turf(source)
```

Comment:
doublechecking, what happens if I throw an item at a thing we could normally pass through that's inside a window/windoor?

Reply:
nvm brain wrong

Reply:
The window/windoor won't block that item because the thrown object would first leave the window/windoor & then check if it can pass the next object/turf it entered into. It's a move then check thing